### PR TITLE
Added unit tests for C create/free decomp routines

### PR DESCRIPTION
**Pull request by MiCurry**
This commit adds unit tests for smiol.c's create and free decomp routines.

### CODE REVIEW

**Comment by mgduda** [comment]
@MiCurry Should we add tests for the case where either `io_elements` or `compute_elements` are NULL pointers in the call to `SMIOL_create_decomp`?

File: smiol_runner.c [hunk 1]
```diff
@@ -346,3 +358,139 @@ int test_open_close(void)
...
+	}
+
+	/* Create and Free Decomp with elements == 1 */
+	fprintf(stderr, "Everything OK (SMIOL_create_decomp) with 1 elements: ");
+	n_compute_elements = 1;
+	n_io_elements = 1;
+	compute_elements = malloc(sizeof(int64_t) * n_compute_elements);
+	io_elements = malloc(sizeof(int64_t) * n_io_elements);
+	decomp = SMIOL_create_decomp(n_compute_elements, n_io_elements,
+									compute_elements, io_elements);
```
Comment (mgduda) [inline smiol_runner.c]: Is there a reason for the line break here and in a few other places, below? It doesn't seem like this shortens the line at all.

Comment (MiCurry) [reply smiol_runner.c]: I thought the same thing as well, but I think there is a problem with how GitHub is displaying the line, which is strange. If you take a look at the line in Vim, its wrapped correctly.

However, now exploring the line with other viewing options, Less and Emacs are displaying the line in the same way GitHub is. So perhaps removing a few tabs on line 416 is needed.

**Comment by MiCurry** [comment]
> @MiCurry Should we add tests for the case where either `io_elements` or `compute_elements` are NULL pointers in the call to `SMIOL_create_decomp`?

Yes, that test would be good to have. I'll add it to this PR. That test is also missing from the Fortran unit tests. I'll add it in another pull request.

**Comment by mgduda** [comment]
@MiCurry Because the dummy arguments to `SMIOLf_create_decomp` do not have the `pointer` attribute, I think the Fortran runtime will generally stop the program execution if we try to pass an unassociated pointer as an actual argument. So, trying to provide unassociated pointers in Fortran may not be a meaningful unit test.

**Comment by MiCurry** [comment]
@mgduda that sounds good to me! I've just pushed an update that includes the tests you requested and a fix for the long broken line.

The long broken line was being created because I had my Vim tabstop set to 4, so there were extra tab characters in to get the line to align correctly. Aligning it with 8 tabs allowed it to display correctly on GitHub.
